components/ParameterForm.vue

```diff
@@ -297,20 +300,20 @@ const submitForm = async () => {
 
   if (response) {
     const { runId } = response;
+    appStore.setCurrentScenario(formData.value as ParameterSet);
     await navigateTo(`/scenarios/${runId}`);
-  }
+  };
 };
 
 // Set fields whose default values are dependent on other fields' values to their defaults.
-const resetAllDependents = () => {
-  paramMetadata.value?.filter((param) => {
-    return param.updateNumericFrom !== undefined;
-  }).forEach(resetParam);
-};
+paramMetadata.value?.filter((param) => {
+  const isDependent = param.updateNumericFrom !== undefined;
+  const shouldBeSetByStore = appStore.currentScenario?.parameters?.[param.id];
+  return isDependent && !shouldBeSetByStore;
+}).forEach(resetParam);
```

Comment:
Why has this been moved to no longer happen in onMounted?

Reply:
Originally it ran in onMounted because it relies on metadata having been fetched. Then I wondered does it really need to be there, and tried moving it outside of onMounted, and letting Mockoon give us a delayed response (1000ms) to the metadata API call to see if it broke anything. That was fine, so I concluded we could run this in the setup script.

However, I don't think moving it out of 'onMounted' gains us much, and I'm happy to move it back.

Reply:
So the parameter form itself isn't loaded until metadata is available anyway? If so, it shouldn't really make a difference - don't mind either way.